Tasks/PublishSymbols/Helpers.ps1

```diff
@@ -276,17 +233,220 @@ function Get-SourceFilePaths {
+
+            # When the build service runs on the same box as the AT, we use localhost
+            # to connect to the AT.  This is not appropriate for storing inside of a PDB that will
+            # be used on another box, so we have to look up a more durable URL.
+            $locationService = $provider.TfsTeamProjectCollection.GetService([Microsoft.TeamFoundation.Framework.Client.ILocationService])
+            # Retrieve a URI to the location service.
+            $provider.PublicCollectionUrl = [string]$locationService.LocationForAccessMapping(
+                [Microsoft.TeamFoundation.ServiceInterfaces]::LocationService,
+                [Microsoft.TeamFoundation.Framework.Common.LocationServiceConstants]::SelfReferenceLocationServiceIdentifier,
+                $locationService.DefaultAccessMapping)
+            # Remove the location service part to form a collection URI.
+            if ($provider.PublicCollectionUrl.EndsWith(
+                    [Microsoft.TeamFoundation.Framework.Common.LocationServiceConstants]::CollectionLocationServiceRelativePath,
+                    [System.StringComparison]::OrdinalIgnoreCase)) {
+                $provider.PublicCollectionUrl = $provider.PublicCollectionUrl.Substring(
+                    0,
+                    $provider.PublicCollectionUrl.Length - [Microsoft.TeamFoundation.Framework.Common.LocationServiceConstants]::CollectionLocationServiceRelativePath.Length)
+            } else {
+                # TODO: SHOULD THIS REVERT TO ENV VAR?
+            }
+
+            $success = $true
+            return New-Object psobject -Property $provider
+        }
+
+        Write-Warning (Get-LocalizedString -Key 'Only TfsGit and TfsVersionControl source providers are supported for source indexing. Repository type: {0}' -ArgumentList $provider)
+        Write-Warning (Get-LocalizedString -Key 'Unable to index sources.')
+        return
+    } finally {
+        if (!$success) {
+            Invoke-DisposeSourceProvider -Provider $provider
+        }
+    }
+}
+
+function Invoke-DisposeSourceProvider {
+    [cmdletbinding()]
+    param($Provider)
+
+    if ($Provider.TfsTeamProjectCollection) {
+        Write-Verbose 'Disposing tfsTeamProjectCollection'
+        $Provider.TfsTeamProjectCollection.Dispose()
+        $Provider.TfsTeamProjectCollection = $null
+    }
+}
+
+function Invoke-IndexSources {
     [cmdletbinding()]
     param(
-        [string]$CollectionUrl = $(throw 'Missing CollectionUrl'),
-        [string]$TeamProjectId = $(throw 'Missing TeamProjectId'),
-        [string]$RepoId = $(throw 'Missing RepoId'),
-        [string]$CommitId = $(throw 'Missing CommitId'),
-        [string]$SourcesRootPath = $(throw 'Missing SourcesRootPath'),
-        [string[]]$SourceFilePaths = $(throw 'Missing SourceFilePaths')
+        [string[]]$SymbolsFilePaths = $(throw 'Missing SymbolsFilePaths'),
+        [switch]$TreatNotIndexedAsWarning
     )
 
+    # Validate at least one symbols file.
```

Comment:
The next three validations were just moved (the third one used to be performed on each temp file - i.e. unnecessary). That's why they are showing as green here.